oidc-configure.sh

```diff
@@ -0,0 +1,117 @@
+#!/bin/bash
+# Add a new realm
+cd /opt/keycloak-4.5.0.Final/bin/
+./kcadm.sh config credentials --server http://localhost:8080/auth --realm master --user admin  --password KEYCLOAKPASS
+./kcadm.sh create realms -s realm=ondemand -s enabled=true -s loginWithEmailAllowed=false -s rememberMe=true
+
+# Configure LDAP
+REALMID=$(./kcadm.sh get realms/ondemand --fields id | egrep -v '{|}' | sed 's/.*id".*:\s*"//g; s/"//g')
+./kcadm.sh create components -r ondemand -s name=ldap -s providerId=ldap -s providerType=org.keycloak.storage.UserStorageProvider -s parentId=$REALMID  -s 'config.importUsers=["false"]'   -s 'config.priority=["1"]' -s 'config.fullSyncPeriod=["-1"]' -s 'config.changedSyncPeriod=["-1"]' -s 'config.cachePolicy=["DEFAULT"]' -s config.evictionDay=[] -s config.evictionHour=[] -s config.evictionMinute=[] -s config.maxLifespan=[] -s 'config.batchSizeForSync=["1000"]'  -s 'config.editMode=["READ_ONLY"]'  -s 'config.syncRegistrations=["false"]'  -s 'config.vendor=["other"]'  -s 'config.usernameLDAPAttribute=["uid"]' -s 'config.rdnLDAPAttribute=["uid"]' -s 'config.uuidLDAPAttribute=["entryUUID"]'  -s 'config.userObjectClasses=["posixAccount"]' -s 'config.connectionUrl=["ldaps://openldap1.infra.osc.edu:636 ldaps://openldap2.infra.osc.edu:636"]' -s 'config.usersDn=["ou=People,ou=hpc,o=osc"]'  -s 'config.authType=["simple"]'  -s 'config.bindDn=["uid=admin,ou=system"]' -s 'config.bindCredential=["secret"]'  -s 'config.useTruststoreSpi=["never"]'  -s 'config.connectionPooling=["true"]' -s 'config.pagination=["true"]'
+
+# Add OnDemand as a client
+CID=$(./kcadm.sh create clients -r ondemand -f /vagrant/ondemand-clients.json -s clientId=localhost  -s enabled=true -s fullScopeAllowed=true -s accessType=confidential -s directAccessGrantsEnabled=false -i)
+
+# Add Custom Theme
+cd ../themes
+curl -LOk  https://github.com/OSC/keycloak-theme/archive/v0.0.1.zip
+unzip v0.0.1.zip
+../bin/kcadm.sh update realms/ondemand -s "loginTheme=keycloak-theme-0.0.1"
+
+# Install dependencies for building mod_auth_openidc
+yum install httpd24-httpd-devel openssl-devel curl-devel jansson-devel pcre-devel autoconf automake
+
+# Install cjose
+cd /opt
+sudo curl -o cjose-0.5.1.tar.gz  https://github.com/zmartzone/mod_auth_openidc/releases/download/v2.3.0/cjose-0.5.1.tar.gz
```

Comment:
cjose is in OnDemand repos: https://yum.osc.edu/ondemand/latest/web/el7/x86_64/cjose-0.5.1-1.el7.x86_64.rpm

If ondemand Yum repo is present can just do `yum install cjose` but should get pulled in if you install `httpd24-mod_auth_openidc`.